Xamarin.Forms.Core/PlatformConfigurationRegistry.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Collections.Generic;
+
+namespace Xamarin.Forms
+{
+	/// <summary>
+	/// Helper that handles storing and lookup of platform specifics implementations
+	/// </summary>
+	/// <typeparam name="TElement">The Element type</typeparam>
+	internal class PlatformConfigurationRegistry<TElement> : IElementConfiguration<TElement>
```

Comment:
We could lazily create these classes it looks like rather than create them eagerly.